src/azure-cli/azure/cli/command_modules/acr/connected_registry.py

```diff
@@ -0,0 +1,494 @@
+    build_token_id
+)
+
+class ConnectedRegistryModes(Enum):
+    MIRROR = 'mirror'
+    REGISTRY = 'registry'
+
+DEFAULT_GATEWAY_SCOPE = ['config/read', 'config/write', 'message/read', 'message/write']
+REPO_SCOPES_BY_MODE = {
+    ConnectedRegistryModes.MIRROR.value: ['content/read', 'metadata/read'],
+    ConnectedRegistryModes.REGISTRY.value: ['content/read', 'content/write', 'content/delete',
+                                            'metadata/read', 'metadata/write']
+}
+SYNC_SCOPE_MAP_NAME = "{}-sync-scope-map"
+SYNC_TOKEN_NAME = "{}-sync-token"
+REPOSITORY = "repositories/"
+GATEWAY = "gateway/"
+
+logger = get_logger(__name__)
+
+
+def acr_connected_registry_create(cmd, # pylint: disable=too-many-locals, too-many-statements
+                                  client,
+                                  registry_name,
+                                  connected_registry_name,
+                                  repositories=None,
+                                  sync_token_name=None,
+                                  client_token_list=None,
+                                  resource_group_name=None,
+                                  mode=None,
+                                  parent_name=None,
+                                  sync_schedule=None,
+                                  sync_message_ttl=None,
+                                  sync_window=None,
+                                  log_level=None,
+                                  sync_audit_logs_enabled=True):
+
+    if bool(sync_token_name) == bool(repositories):
+        raise CLIError("usage error: you need to provide either --sync-token-name or --repository, but not both.")
+    registry, resource_group_name = get_registry_by_name(cmd.cli_ctx, registry_name, resource_group_name)
+    subscription_id = get_subscription_id(cmd.cli_ctx)
+
+    if not registry.data_endpoint_enabled:
+        logger.warning("Registry '%s' dedicated data endpoint is disabled. Enabling...", registry_name)
+        from .custom import acr_update_custom, acr_update_get, acr_update_set
+        acr_update_set(cmd, cf_acr_registries(cmd.cli_ctx), registry_name, resource_group_name,
+                       parameters=acr_update_custom(cmd, acr_update_get(cmd), data_endpoint_enabled=True))
+        logger.info("Registry '%s' dedicated data endpoint successfully enabled.", registry_name)
+
+    ErrorResponseException = cmd.get_models('ErrorResponseException')
+    parent = None
+    if parent_name:
+        try:
+            parent = acr_connected_registry_show(cmd, client, parent_name, registry_name, resource_group_name)
+        except ErrorResponseException as ex:
+            if ex.response.status_code == 404:
+                raise CLIError("The parent connected registry '{}' could not be found.".format(parent_name))
+            raise CLIError(ex)
+        if parent.mode.lower() != ConnectedRegistryModes.REGISTRY.value and parent.mode.lower() != mode.lower():
```

Comment:
should the parent mode be same as the parent?

Reply:
when the parent is mirror the child can't be registry. Since the parent isn't going to have the permissions the child requires.